bot.py

```diff
@@ -133,6 +166,144 @@ async def info(self, ctx: commands.Context[Any]) -> None:
         await ctx.reply(embed=constants.INFO_REPLY)
 
 
+P = ParamSpec("P")
+T = TypeVar("T")
+
+
+def only_from_guilds(*servers: list[int]) -> Callable[[Callable[P, T]], Callable[P, T]]:
```

Comment:
The idea was to make the decorator usable with both a manually-typed list of arguments and also a list passed in. If we follow your suggestion, wouldn't the manually-typed case want to type out a tuple, not a list? In order to avoid issues with statically storing mutable data?

Reply:
oh right, i forgot to mention but i think we should be converting it to a set anyways, since its doing a O(n) lookup currently, that will also solve the mutability issue even if we take a list

also, if we want to take either `list[list[int]] | list[int]` we can just annotate it as `list[int] | int`